cassandra/io/libevreactor.py

```diff
@@ -310,22 +310,28 @@ def handle_write(self, watcher, revents, errno=None):
                 with self._deque_lock:
                     next_msg = self.deque.popleft()
             except IndexError:
+                if not self._socket_writable:
+                    self._socket_writable = True
                 return
 
             try:
                 sent = self._socket.send(next_msg)
             except socket.error as err:
                 if (err.args[0] in NONBLOCKING or
                         err.args[0] in (ssl.SSL_ERROR_WANT_READ, ssl.SSL_ERROR_WANT_WRITE)):
+                    if err.args[0] in NONBLOCKING:
+                        self._socket_writable = False
                     with self._deque_lock:
                         self.deque.appendleft(next_msg)
                 else:
                     self.defunct(err)
                 return
             else:
                 if sent < len(next_msg):
+                    self._socket_writable = False
```

Comment:
Do you think it's too aggressive? should I do this only if `send == 0`?

Reply:
I'm not too familiar with the low level socket API but IIRC if `sent == 0` that would mean that the remote endpoint closed the connection right? So you would want to close the socket instead of setting the flag to false in this scenario.

I'm not sure if setting the flag immediately here is better than letting the loop continue and potentially receive a `EAGAIN` / `EWOULDBLOCK` in the next `send()`... I think that this is fine and maybe better than letting the loop continue.

When is the `writable` flag being set back to `true` again? Should the driver check if it is `false` after a `socket.send()` succeeds with all data being written and set it to `true`?

Reply:
oh I just read this on the PR description:

> When the send buffer is writable again, we flush all the current write queue then mark the connection as writable again.

This answers my question related to setting the `writable` flag 👍

Reply:
Good point... I think it's not really adding any value. I will remove the sent check and wait egain/ewouldblock.